library/core/src/ptr/iter.rs

```diff
@@ -0,0 +1,513 @@
+use crate::fmt::{self, Debug};
+use crate::hash::{Hash, Hasher};
+use crate::iter::{FusedIterator, TrustedLen, TrustedRandomAccess, TrustedRandomAccessNoCoerce};
+use crate::mem;
+
+macro_rules! ptr_iter {
+    ($ty:ident, *$mutability:tt T) => {
+        /// Iterator with items generated by incrementing a raw pointer.
+        ///
+        /// # Safety
```

Comment:
I think, we should note that `T` must not be ZST because in that case it is very easy to make invalid iterations.